### DISCUSSION TITLE
KeyError while rendering page: 0.36.0

### DISCUSSION THREAD

**Discussion by jkrobicki** [idom-team/idom#639] [Problem]
KeyError appers when I try to display components, however it works with version 0.34.0

![image](https://user-images.githubusercontent.com/88193723/152132499-229ebaae-c9af-499c-acd2-0fd036a7cf6e.png)

**Comment by rmorshea** [comment]
Could you provide a minimal example to reproduce this? That would help a lot in tracking down the bug.

**Comment by jkrobicki** [comment]
Here is the code replicating the problem. Blank page appears when clicking the button. It's working on 0.34.0.
```python
from idom import html, run, use_state, component, event, vdom, EventHandler
from sanic import Sanic
from pathlib import Path
from idom.server.sanic import PerClientStateServer

@component
def show_title1(current_page, set_current_page):
    def handle_click(event):
        set_current_page("Epics")

    return html.button({"onClick": handle_click}, "Bird")


@component
def page():

    current_page, set_current_page = use_state("Users")
    pages = ["Users", "Timelogs", "Epics", "Clients"]

    timelogs_page = html.h1(
        {"class": "text-white"}, "Timelogs Page, not implemented yet"
    )
    epics_page = html.h1({"class": "text-white"}, "Epics Page, not implemented yet")

    # clients_page = html.h1({"class": "text-white"}, "Clients Page, not implemented yet")

    print("here", current_page)
    # current_page_component = html.div()
    if current_page == "Users":
        current_page_component = show_title1(current_page, set_current_page)
    elif current_page == "Epics":
        current_page_component = epics_page
    elif current_page == "Timelogs":
        current_page_component = timelogs_page
    elif current_page == "Clients":
        current_page_component = clients_page()
    else:
        current_page_component = html.h1(
            {"class": "text-white"}, "Test Page, not implemented yet"
        )

    return html.div({"class": "flex w-full"}, current_page_component)


app = Sanic(__name__)
HERE = Path(__file__).parent
app.static("/static", str(HERE / "tailwind/build"))
PerClientStateServer(
    page,
    {
        "redirect_root_to_index": False,
    },
    app,
)


def run():
    app.run(
        host="0.0.0.0",
        port=8001,
        workers=1,
        debug=True,
    )


run()
```
and same error appears:

![image](https://user-images.githubusercontent.com/88193723/152190485-5b9acc48-47ee-4ac6-9191-e38db07bbdbd.png)

**Reply by rmorshea** [reply]
Ok, found the bug and thankfully there's a simple fix. I'll push a release out this evening with the fix.

**Reply by rmorshea** [reply]
This little bug is yet more motivation to work on https://github.com/idom-team/idom/issues/624

**Reply by Archmonger** [reply]
Agreed. I personally had trouble mentally tracing the call stack within layout.py.

I'd still recommend targeting for v2 though, since v1 is getting a bit overloaded.

**Reply by rmorshea** [reply]
Yeah, definitely gonna keep this for v2.

**Comment by rmorshea** [comment]
This should be fixed in https://github.com/idom-team/idom/discussions/643:
![Peek 2022-02-02 18-58](https://user-images.githubusercontent.com/4041990/152274301-0b57f7ea-79ed-4e80-b53c-f80ac8c2a0e9.gif)

**Reply by jkrobicki** [reply]
Indeed, the issue is solved on my end too with 0.36.1

**Reply by rmorshea** [reply]
be sure to bump to 0.36.2 which includes a hotfix for something else.
